### PR TITLE
Variable Select Input

### PR DESCRIPTION
**Pull request by schloerke**
For use within tidy eval.

Kitchen Variable Select Input Example: https://gist.github.com/schloerke/580ee1587e58ac1f367f00bb1acc22fc
Link to app: https://beta.rstudioconnect.com/connect/#/apps/3782/access

Two small examples:
```r
shinyApp(
  ui = fluidPage(
    varSelectInput("variable", "Variable:", mtcars),
    plotOutput("data")
  ),
  server = function(input, output) {
    output$data <- renderPlot({
      ggplot2::qplot(!!input$variable, data = mtcars)
    })
  }
)

shinyApp(
  ui = fluidPage(
    varSelectInput("variable", "Variable:", mtcars, multiple = TRUE),
    tableOutput("data")
  ),
  server = function(input, output) {
    output$data <- renderTable({
       if (length(input$variable) == 0) return(mtcars)
       mtcars %>% dplyr::select(!!!input$variable)
    }, rownames = TRUE)
  }
)
```

### CODE REVIEW

**Comment by wch** [comment]
Do you mind deploying the app to rsc.radixu.com, or beta.rstudioconnect.com?

File: R/input-select.R [hunk 1]
```diff
@@ -212,3 +212,121 @@ selectizeIt <- function(inputId, select, options, nonempty = FALSE) {
...
+
+
+
+
+
+
+
+#' Create a select list input control from a data.frame
+#'
+#' Create a select list that can be used to choose a single or multiple items
```
Comment (hadley) [inline R/input-select.R]: This seems to be a duplicate of the title? I think you can omit

File: R/input-select.R [hunk 2]
```diff
@@ -212,3 +212,121 @@ selectizeIt <- function(inputId, select, options, nonempty = FALSE) {
...
 }
+
+
+
+
+
+
+
+
+#' Create a select list input control from a data.frame
```
Comment (hadley) [inline R/input-select.R]: Maybe "Select variables from a data frame" ?

File: R/input-select.R [hunk 3]
```diff
@@ -212,3 +212,121 @@ selectizeIt <- function(inputId, select, options, nonempty = FALSE) {
...
+#'
+#' Create a select list that can be used to choose a single or multiple items
+#' from the column names of a data frame.
+#'
+#' The resulting output value will be returned as a symbol or a list of symbols
```
Comment (hadley) [inline R/input-select.R]: ... returned as a symbol (if multiple = FALSE) or a list of symbols (if multiple = TRUE)...

Or maybe make a bulleted list? Then you could explain `!!` vs `!!!` for each bullet too.

File: R/input-select.R [hunk 4]
```diff
@@ -212,3 +212,121 @@ selectizeIt <- function(inputId, select, options, nonempty = FALSE) {
...
+#'
+#' # single selection
+#' shinyApp(
+#'   ui = fluidPage(
+#'     varSelectInput("variable", "Variable:", mtcars),
+#'     plotOutput("data")
+#'   ),
+#'   server = function(input, output) {
+#'     output$data <- renderPlot({
+#'       ggplot2::qplot(!!input$variable, data = mtcars)
```
Comment (hadley) [inline R/input-select.R]: I'd prefer to avoid `qplot()` if possible.

**Comment by schloerke** [comment]
Thanks @hadley for the docs check! (updated details, title, and changed `qplot` to `ggplot`)

File: R/input-select.R [hunk 5]
```diff
@@ -212,3 +212,130 @@ selectizeIt <- function(inputId, select, options, nonempty = FALSE) {
...
+    choices = choices,
+    selected = selected,
+    multiple = multiple,
+    selectize = selectize,
+    width = width,
+    size = size
+  )
+
+  # set the select tag class to be "symbol"
+  selectAttribs <- selectInputVal$children[[2]]$children[[1]]$attribs
```
Comment (jcheng5) [inline R/input-select.R]: Can you add a check here that `selectInputVal$children[[2]]$children[[1]]` is actually a `<select>`? (Either here, or in a unit test--and if a unit test, both `selectize=TRUE` and `FALSE`.) I just want to be sure we catch it if we slightly change the DOM structure of `selectInput`.

File: DESCRIPTION [hunk 6]
```diff
@@ -90,8 +90,9 @@ Suggests:
 URL: http://shiny.rstudio.com
 BugReports: https://github.com/rstudio/shiny/issues
 Remotes:
+    tidyverse/ggplot2,
```
Comment (jcheng5) [inline DESCRIPTION]: Does this mean we wouldn't be able to release Shiny to CRAN without a ggplot2 release first? If so, any idea of a timeline @hadley?

Comment (schloerke) [reply DESCRIPTION]: to work with the `aes()` I believe we'll have to wait.  But that's only for the example.  

If there is a different example, we could use it instead to release sooner.

File: R/input-select.R (hunk 5 above)
Comment (schloerke) [reply R/input-select.R]: I'll add a unit test

File: DESCRIPTION (hunk 6 above)
Comment (hadley) [reply DESCRIPTION]: We're in the middle of the release process, with submission scheduled for June 25.

Comment (hadley) [reply DESCRIPTION]: (But equally it would be fine to eliminate that example — you can't replace it with anything particularly useful in released ggplot2)

**Pull request merged**